HSMP.hpp

```diff
@@ -231,6 +292,28 @@ std::shared_ptr<ClientRequest> ProcessRequest(std::string buffer) {
       return mreq;
     }
 
+    case 'b': {
+      auto breq = std::make_shared<BroadcastRequest>();
+
+      breq->mensajeTamano = atoi(buffer.substr(1, 3).c_str());
+
+      breq->mensajeContenido = buffer.substr(4, breq->mensajeTamano);
+      return breq;
+    }
+
+    case 'u': {
+      auto ureq = std::make_shared<UploadFileRequest>();
+
+      ureq->fileNombreTamano = atoi(buffer.substr(1, 3).c_str());
+      ureq->fileDataTamano = atoi(buffer.substr(4, 10).c_str());
+      ureq->destinatarioTamano = atoi(buffer.substr(14, 2).c_str());
+
+      ureq->fileNombreContenido = buffer.substr(16, ureq->fileNombreTamano);
+      ureq->fileDataContenido; // depende el archivo
+      ureq->destinatarioContenido = buffer.substr(16 + ureq->fileNombreTamano + ureq->fileDataTamano, ureq->destinatarioTamano);
```

Comment:
UploadFileRequest no tiene atributos con estos nombres.

Reply:
fixed this bugs.